apps/analysis_framework/serializers.py

```diff
@@ -708,3 +708,36 @@ def create(self, validated_data):
         validated_data['framework'] = self.framework
         validated_data['added_by'] = self.context['request'].user
         return super().create(validated_data)
+
+
+class AnalysisFrameworkCloneGlSerializer(serializers.Serializer):
+    title = serializers.CharField(required=True)
+    description = serializers.CharField(required=False)
+    project = serializers.IntegerField(required=False)
+
+    class Meta:
+        model = AnalysisFramework
+        fields = ('title', 'description', 'project')
+
+    def validate(self, validated_data):
+        project_id = validated_data.get('project')
+        # Check if project exists and user has access to it
+        if project_id is not None:
+            project = Project.objects.filter(id=project_id).first()
+            if project is None:
+                raise serializers.ValidationError('Invalid project ID')
+            if not project.can_modify(self.context['request'].user):
+                raise serializers.ValidationError('User does not have permission to modify the project')
+            validated_data['project'] = project
+        return validated_data
+
+    def create(self, validated_data):
+        af = self.context['request'].active_af
+        new_af = af.clone(self.context['request'].user, validated_data)
+        new_af.add_member(self.context['request'].user, new_af.get_or_create_owner_role())
+
+        if project := validated_data.get('project'):
+            project.analysis_framework = new_af
+            project.modified_by = self.context['request'].user
+            project.save(update_fields=['analysis_framework', 'modified_by'])
```

Comment:
Can you confirm if modified_at also updates?

Reply:
Yes, Modified_at field is working fine.

Reply:
Check if `modified_at` is also updated. Or let's just add that field to the update_fields for visibility

Reply:
Checked! `Modified_at` is also updated.